### PR TITLE
Changelog updated.

### PR DESCRIPTION
**Pull request by andreitraistaru**
Signed-off-by: Trăistaru Andrei Cristian <atc@amazon.com>

## Reason

Keep track of Rust toolchain upgrade in CHANGELOG.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] New `unsafe` code is documented.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md

### CODE REVIEW

**Review by zulinx86 — APPROVED** [review]

**Review by zulinx86 — COMMENTED** [review]
Could you mention the commit that upgraded it to 1.64.0 in this PR or the commit message?

**Review by zulinx86 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,7 @@
 - Changed the jailer option `--exec-file` to fail if the filename does not
   contain the string `firecracker` to prevent from running non-firecracker
   binaries.
+- Rust toolchain upgraded to 1.64.0.
```
Comment (xmarcalx) [inline CHANGELOG.md:26]: let's add also: from x to y

**Review by xmarcalx — CHANGES_REQUESTED** [review]

**Review by zulinx86 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (xmarcalx) [reply CHANGELOG.md]: thanks a lot!

**Review by xmarcalx — APPROVED** [review]
LGTM, thanks a lot

**Pull request merged**